### DISCUSSION TITLE
storybook v6 build-storybook doesn't work

### DISCUSSION THREAD

**Discussion by liwang2002** [storybookjs/storybook#16594] [Help] [answered]
I upgraded storybook to v6 with main.js and preview.js etc. It works in local when I run "npm run storybook", the storybook components display OK. But when I do deploy and publish, the page shows "The glob specified in main.js isn't correct.
No stories are defined in your story files.".  The build command I used:

npm run build-storybook -- -o ./build
npx http-server .build

my node version is v14.17.6, npm version is 6.14.15. my main.js is:
```
module.exports = {
    stories: ['../**/*.stories.@js'],
    addons: [
        '@storybook/addon-actions',
        '@storybook/addon-links',
        '@storybook/addon-knobs',
        '@storybook-readme',
        '@storybook/addon-backgrounds',
        '@storybook/addon-a11y',
        '@storybook/addon-storysource'
    ],
    refs: {
        'design-system': {
          title: "Storybook component library V6",
          url: "https://some-url"
        }
       }
}
```

**Comment by shilman** [comment, marked as answer]
Try:

```
stories: ['../**/*.stories.js'],
```

**Reply by liwang2002** [reply]
Thanks shilman! removed@, same error

**Reply by shilman** [reply]
Do you a have a reproduction repo you can share? If not, can you create one? See [how to create a repro](https://storybook.js.org/docs/react/contribute/how-to-reproduce).

**Reply by liwang2002** [reply]
Hi shilman, Thanks. but not sure I could share production code for security reason, my local working when I do npm run storybook. the main.js seems ok.

**Reply by liwang2002** [reply]
when I run storybook locally, it works fine, it seems for me that the main.js is OK and it find the .stories all components, but why after build, it doesn't show the components? seem to me, the build command doesn't pack the storybook correctly or missing something? or did I used correct command for build?

**Reply by shilman** [reply]
which version of storybook are you using?

**Reply by liwang2002** [reply]
Storybook 6.3.12

**Reply by liwang2002** [reply]
storybook v6 works now, the reason is the @js format was wrong. thanks
